src/hints/block_context.rs

```diff
@@ -176,6 +147,141 @@ pub fn load_deprecated_class(
+
+    bytecode_segment_structure = create_bytecode_segment_structure(
+        bytecode=compiled_class.bytecode,
+        bytecode_segment_lengths=compiled_class.bytecode_segment_lengths,
+        visited_pcs=compiled_class_visited_pcs[compiled_class_hash],
+    )
+
+    cairo_contract = get_compiled_class_struct(
+        identifiers=ids._context.identifiers,
+        compiled_class=compiled_class,
+        bytecode=bytecode_segment_structure.bytecode_with_skipped_segments()
+    )
+    ids.compiled_class = segments.gen_arg(cairo_contract)"#
+};
+pub fn load_class_inner(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let class_iter = exec_scopes.get_mut_ref::<IntoIter<Felt252, CasmContractClass>>("compiled_class_facts")?;
+
+    let (compiled_class_hash, class) = class_iter.next().unwrap();
+
+    exec_scopes.insert_value("compiled_class_hash", compiled_class_hash);
+    exec_scopes.insert_value("compiled_class", class.clone()); //TODO: is this clone necessary?
+
+    // TODO: implement create_bytecode_segment_structure (for partial code loading)
+
+    let class_base = vm.add_memory_segment();
+    write_class(vm, class_base, class.clone())?; //TODO: clone unnecessary
+
+    insert_value_from_var_name("compiled_class", class_base, vm, ids_data, ap_tracking)
+}
+
+pub const BYTECODE_SEGMENT_STRUCTURE: &str = indoc! {r#"
+    vm_enter_scope({
+        "bytecode_segment_structure": bytecode_segment_structure
+    })"#
+};
+pub fn bytecode_segment_structure(
+    _vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    _ids_data: &HashMap<String, HintReference>,
+    _ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    // TODO: implemente when partial code loading is implemented
+    exec_scopes.enter_scope(HashMap::default());
+    Ok(())
+}
+
+pub const LOAD_CLASS: &str = indoc! {r#"
+    computed_hash = ids.compiled_class_fact.hash
+    expected_hash = compiled_class_hash
+    assert computed_hash == expected_hash, (
+        "Computed compiled_class_hash is inconsistent with the hash in the os_input. "
+        f"Computed hash = {computed_hash}, Expected hash = {expected_hash}.")
+
+    vm_load_program(
+        compiled_class.get_runnable_program(entrypoint_builtins=[]),
+        ids.compiled_class.bytecode_ptr
+    )"#
+};
+pub fn load_class(
+    _hint_processor: &dyn HintProcessor,
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<HintExtension, HintError> {
+    let compiled_class_fact_addr = get_relocatable_from_var_name("compiled_class_fact", vm, ids_data, ap_tracking)?;
+    let computed_hash = vm.get_integer(compiled_class_fact_addr)?; //TODO: calculate .hash explicitly
```

Comment:
Nitpick: Shouldn't you define `compiled_class_fact` as a struct that derives `CairoType`?